backend/de.metas.acct.base/src/main/sql/postgresql/ddl/functions/report_M_Product_QtyBooked_For_Date.sql

```diff
@@ -50,30 +49,37 @@ WITH tmp_costdetails AS
              WHERE TRUE
                AND cd.m_costelement_id = p_M_CostElement_ID
                AND COALESCE(cd.m_inoutline_id, cd.m_movementline_id, cd.m_inventoryline_id) IS NOT NULL
+               AND (p_M_Product_ID <= 0 OR p_M_Product_ID = cd.m_product_id)
+               AND (
+                     (p_date IS NULL AND COALESCE(inv.movementdate, io.MovementDate, m.movementdate) <= NOW())
```

Comment:
but why enforcing NOW()?

Reply:
Right. The way you suggested in your proto is better than mine so I replaced. Please, refresh